lessons/lesson5/lesson.md

```diff
@@ -0,0 +1,124 @@
+### Writing docstrings
+
+We'll continue with the webserver library from the exercise.
+
+%(example1/example1.jsonnet)s
+
+This library provides a number of functions to create a webserver. In docsonnet lingo this
+is called a 'package'. Each function has a few arguments which usually have a fixed type,
+as Jsonnet itself is not a typed language we'll simply try to document the type.
+
+---
+
+To get started, let's grab docsonnet's doc-util library. It provides a few functions to
+write consistent docstrings for the different objects in Jsonnet.
+
+`$ jb install github.com/jsonnet-libs/docsonnet/doc-util`
+
+Note that the [docs for
+doc-util](https://github.com/jsonnet-libs/docsonnet/blob/master/doc-util/README.md) are
+also rendered by docsonnet.
+
+---
+
+%(example1/example2.jsonnet)s
+
+The docsonnet puts a claim on keys that start with a hash `#`. It assumes that keys
+starting with a hash symbol is very uncommon, additionally it closely relates to how
+comments are written.
+
+The package definition is put in the `#` key without a value.
+
+The `url` argument is the part that can be used to install this library with
+jsonnet-bundler:
+
+`$ jb install <url>`
+
+In combination with the `url`, the `filename` refers to what should be imported, note the
+neat `std.thisFile` shortcut so we don't have to remember to change the filename here if
+we do so.
```

Comment:
"if we do so". I don't really understand this sentence

Reply:
```suggestion
neat `std.thisFile` shortcut so we don't have to remember to change the filename here if
we ever rename the file.
```